### ISSUE TITLE
Arcane Sanctuary - Killing ghosts on my way to Summoner

### ISSUE THREAD

**Issue by Rakkah** [kolton/d2bot-with-kolbot#592]
Hey,

I wanted to edit Countess.js by myself and add a function from it to Summoner.js. If I set a command to true, it'll clear all ghosts on my way to Summoner, like it is in Countess.js. But I don't really understand the code in Countess.js, I mean, even the statement Config.Countess.KillGhosts doesn't exist in the script, so I have no idea how does it clear ghosts in Tower.

And please, don't blame my JavaScript knowledge if that's an easy question. :(

**Comment by gogos-venge** [comment]
Yeah your frustration is justified. That is because Countess.js is not actually doing the clearing of ghosts directly. Instead, Pather.js does it. At line 24 in Pather.js
```
if (Config.Countess.KillGhosts && [21, 22, 23, 24, 25].indexOf(me.area) > -1) {
	monList = Attack.getMob(38, 0, 30);

	if (monList) {
		Attack.clearList(monList);
	}
}
```
You can clearly see Config.Countess.KillGhosts boolean is used here along with the corresponding arreas of Tower 1,2,3 etc. In order to kill the ghosts in the Arcane Sanctuary copy and paste the Tower Cellar code I pasted above just below it in Pather.js and then go to the sdk folder of kolton and:
1) Find the ghosts id in arcance sanctuary and change `Attack.getMob(.....)`
2) Find the sanctuary area to change this `[21, 22, 23, 24, 25].indexOf(me.area) > -1` into something like this: `me.area === <sanctuary area>`

Regards, and best of luck.

**Comment by gogos-venge** [comment]
Sorry to spoil the solution but I liked your idea of clearing ghosts in arcane, and got it done:
Pather.js
```
if (Config.Summoner.KillGhosts && me.area === 74) {
	monList = Attack.getMob(40, 0, 30);

	if (monList) {
		Attack.clearList(monList);
	}
}
```

Your char config:
```
Scripts.Summoner = true;
		Config.Summoner.FireEye = false;
		Config.Summoner.KillGhosts = true;
```

Seems to work with no issues ATM. Please make some tests and tell me of any bugs.

**Comment by gogos-venge** [comment]
Yeah it **should** be like this way. If you are looking for a script that clears all Arcane Sanctuary, then you should be modifying the ClearAnyArea in area 74, to attack just unit 40 which is the Specter. Otherwise Summoner script just travels the way Summoner is located, and occasionally kill Specters if stumbled upon them. It's also how Countess' one works. It won't clear any room of ghosts, only the ones in a range of 30 yards between the paths of each area (door to door)

**Comment by adamveg** [comment]
> 
> 
> Sorry to spoil the solution but I liked your idea of clearing ~ghosts~ specters in arcane, and got it done:
> Pather.js (under line 30)
> 
> ```js
> if (Config.Summoner.KillSpecters && me.area === 74) {
> 	monList = Attack.getMob(40, 0, 30);
> 
> 	if (monList) {
> 		Attack.clearList(monList);
> 	}
> }
> ```
> 
> Your char config:
> 
> ```js
> Scripts.Summoner = true;
> 		Config.Summoner.FireEye = false;
> 		Config.Summoner.KillSpecters = true;
> ```
> 
> Seems to work with no issues ATM. Please make some tests and tell me of any bugs.

Did you update this confing ? Because it doesn't work in my panter etc.
